inference/core/data_models.py

```diff
@@ -699,3 +713,92 @@ class SamSegmentationResponse(BaseModel):
+    """
+
+    x: float = Field(description="The center x-axis pixel coordinate of the prediction")
+    y: float = Field(description="The center y-axis pixel coordinate of the prediction")
+    width: float = Field(
+        description="The width of the prediction bounding box in number of pixels"
+    )
+    height: float = Field(
+        description="The height of the prediction bounding box in number of pixels"
+    )
+    confidence: float = Field(
+        description="The detection confidence as a fraction between 0 and 1"
+    )
+    class_name: str = Field(alias="class", default="face", description="The predicted class label")
+
+    landmarks: Union[List[Point], List[Point3D]]
+
+
+class GazeDetectionPrediction(BaseModel):
+    """Gaze Detection prediction.
+
+    Attributes:
+        face (FaceDetectionPrediction): The face prediction.
+        pitch (float): Pitch (degree) of the detected face.
+        yaw (float): Yaw (degree) of the detected face.
+    """
+
+    face: FaceDetectionPrediction
+
+    pitch: float = Field(description="Pitch (degree) of the detected face")
+    yaw: float = Field(description="Yaw (degree) of the detected face")
+
+
+class GazeDetectionInferenceRequest(BaseModel):
+    """Request for gaze detection inference.
+
+    Attributes:
+        api_key (Optional[str]): Roboflow API Key.
+        gaze_version_id (Optional[str]): The version ID of Gaze to be used for this request.
+        is_cropped_face_image (Optional[bool]): If false, face detection will be applied; if true, face detection will be ignored and the whole input image will be used for gaze detection.
+        image (Union[List[InferenceRequestImage], InferenceRequestImage]): Image(s) for inference.
+    """
+
+    api_key: Optional[str] = ApiKey
+    gaze_version_id: Optional[str] = Field(
+        default=GAZE_VERSION_ID,
+        example="l2cs",
+        description="The version ID of Gaze to be used for this request. Must be one of l2cs.",
+    )
+
+    is_cropped_face_image: Optional[bool] = Field(
+        default=False,
+        example=False,
+        description="If false, face detection will be applied; if true, face detection will be ignored and the whole input image will be used for gaze detection",
+    )
+
+    image: Union[List[InferenceRequestImage], InferenceRequestImage]
+
+
+class GazeDetectionInferenceResponse(BaseModel):
+    """Response for gaze detection inference.
+
+    Attributes:
+        predictions (List[GazeDetectionPrediction]): List of gaze detection predictions.
+        time_total (float): The total processing time (second).
+        time_load_img (float): The processing time for loading image (second).
+        time_face_det (float): The processing time for face detection (second).
+        time_gaze_det (float): The processing time for gaze detection (second).
+    """
+
+    predictions: List[GazeDetectionPrediction]
+
+    time_total: float = Field(description="The total processing time (second)")
```

Comment:
Change `total_time` to `time` to better match other response models

Reply:
How about we use the following definition:
time: Union[float, List[float]]
a. in production, we only return a single float for the processing time;
b. when necessary in debug, we can return a list of processing times for each crucial step

Reply:
Let's keep `time` just a `float` to match other model responses. You could make the other time fields optional and only return them in debug mode if you want to keep them around. But we don't currently have any sort of debug mode options, so might be better to just remove them for now.